### PR TITLE
fix(publish): add node specifiers

### PR DESCRIPTION
**Pull request by crowlKats**

### CODE REVIEW

**Review by dsherret — APPROVED** [review]
LGTM!

**Review by bartlomieju — COMMENTED** [review]
Can we please test is somehow?

**Review by lucacasonato — COMMENTED** [review]
Can you add a test?

**Comment by dsherret** [comment]
Hey Leo, can you test this please?

File: cli/tests/integration/publish_tests.rs [hunk 1]
```diff
@@ -94,6 +94,14 @@ itest!(successful {
   http_server: true,
 });
 
+itest!(node_specifier {
+  args: "publish --token 'sadfasdf'",
+  output: "publish/node_specifier.out",
+  cwd: Some("publish/node_specifier"),
+  envs: env_vars_for_registry(),
```
Comment (dsherret) [inline cli/tests/integration/publish_tests.rs:101]: Need to also add env vars for local npm registry here too.

**Pull request merged**